services/httpd/handler_test.go

```diff
@@ -284,6 +284,65 @@ func TestHandler_Query_ErrResult(t *testing.T) {
 	}
 }
 
+// Ensure the handler handles ping requests correctly.
+func TestHandler_Ping(t *testing.T) {
+	h := NewHandler(false)
+	w := httptest.NewRecorder()
+	h.ServeHTTP(w, MustNewRequest("GET", "/ping", nil))
+	if w.Code != http.StatusNoContent {
+		t.Fatalf("unexpected status: %d", w.Code)
+	}
+	h.ServeHTTP(w, MustNewRequest("HEAD", "/ping", nil))
+	if w.Code != http.StatusNoContent {
+		t.Fatalf("unexpected status: %d", w.Code)
+	}
+}
+
+// Ensure the handler handles ping requests correctly, when waiting for leader.
+func TestHandler_PingWaitForLeader(t *testing.T) {
+	h := NewHandler(false)
+	w := httptest.NewRecorder()
+	h.ServeHTTP(w, MustNewRequest("GET", "/ping?wait_for_leader=1s", nil))
```

Comment:
yup, all good by me.  If there was a spec it would be different, but because there is not, consistency is the only thing that really matters.